cdk/api.py

```diff
@@ -58,16 +59,17 @@ def __init__(self, scope: core.Construct, id: str,
             ]
         )
 
-        on_demand_function.add_environment(
+        on_demand_function.function.add_environment(
             "WEBHOOK_ENDPOINT_URL",
-            (f"https://{self.api.rest_api_id}.execute-api.{self.region}"
-             f".amazonaws.com/{self.lambda_release_alias}/new_recording")
+            (f"https://{self.api.rest_api_id}.execute-api.{core.Stack.of(self).region}"
+             f".amazonaws.com/{lambda_release_alias}/new_recording")
         )
 
     def add_monitoring(self, monitoring):
 
-        for metric_name in ["4XXError", "5XXError"]:
+        for error_type in ["4XX", "5XX"]:
             for resource in [self.new_recording_resource, self.ingest_resource]:
+                metric_name = f"{resource.path.strip('/')}{error_type}Error"
                 alarm = cloudwatch.Alarm(self, f"{metric_name}Alarm",
                     metric=cloudwatch.Metric(
                         metric_name=metric_name,
```

Comment:
In our current setup we have one set of 4xx and 5xx alarms for the whole api... I like the idea of having them per resource... but there is an error if they don't have unique names. Also, I think I would prefer to not have the 4xx alarm for the ingest endpoint since those error emails are just people putting the incorrect link into the Zoom+ tool.

Reply:
hmm, i agree we don't want an alert every time that happens, but now i'm wondering what kind of 4xx errors do we get for the webhook endpoint? also if I could stop trying to be such a HTTP purist would it make sense to have the on-demand function return something besides a 4xx in those cases of producer error? nah forget it. IIRC we'd have to make a change on the opencast side too. 

A better question is, is there a reason to get an alert for **any** 4xx for either endpiont?